### PR TITLE
plugin/health: add lameduck mode

### PR DESCRIPTION
**Pull request by miekg**
Add a way to configure lameduck more, i.e. set health to false, stop
polling plugins. Then wait for a duration before shutting down. As the
health middleware is configured early on in the plugin list, it will
hold up all other shutdowns, meaning we still answer queries.

### CODE REVIEW

**Comment by miekg** [comment]
Fixes  #1329

**Comment by miekg** [comment]
~~~
.:9853
2018/01/11 16:23:51 [INFO] CoreDNS-1.0.3
2018/01/11 16:23:51 [INFO] linux/amd64, go1.9.2, 
CoreDNS-1.0.3
linux/amd64, go1.9.2, 
::1 - [11/Jan/2018:16:24:21 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 116.101494ms
::1 - [11/Jan/2018:16:24:22 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 691.294µs
::1 - [11/Jan/2018:16:24:23 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 273.097µs
::1 - [11/Jan/2018:16:24:24 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 226.296µs
2018/01/11 16:24:25 [INFO] SIGINT: Shutting down
2018/01/11 16:24:25 [INFO] Going into lameduck mode for 10s
::1 - [11/Jan/2018:16:24:25 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 243.527µs
::1 - [11/Jan/2018:16:24:26 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 286.193µs
::1 - [11/Jan/2018:16:24:27 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 236.558µs
::1 - [11/Jan/2018:16:24:28 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 239.255µs
::1 - [11/Jan/2018:16:24:29 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 273.277µs
::1 - [11/Jan/2018:16:24:30 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 203.564µs
::1 - [11/Jan/2018:16:24:31 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 334.589µs
::1 - [11/Jan/2018:16:24:32 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 511.567µs
::1 - [11/Jan/2018:16:24:33 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 229.837µs
::1 - [11/Jan/2018:16:24:34 +0000] "MX IN miek.nl. udp 37 false 4096" NOERROR qr,rd,ra 421 293.179µs
~~~

File: plugin/health/README.md [hunk 1]
```diff
@@ -21,6 +21,17 @@ a 503. *health* periodically (1s) polls plugin that exports health information.
...
+More options can be set with this extended syntax:
+
+~~~
+health [ADDRESS] {
+    lameduck DURATION
+}
+~~~
+
+* Where `lameduck` will make the process unhealthy then *wait* for **DURATION** before the process
+  shuts down. If you want to explicitaly disable use `lameduck 0`.
```
Comment (johnbelamaric) [inline plugin/health/README.md]: why would you need to explicitly disable? is it enabled by default?

Comment (miekg) [reply plugin/health/README.md]: No. I'll delete this; obviously confusing.

**Comment by miekg** [comment]
Matt will merge https://github.com/mholt/caddy/pull/1995 in one form or another. As reminder we should update the caddy version in `Makefile` when he does.

**Comment by miekg** [comment]
caddy now does it on sigterm as well, but we don't have to wait for a caddy release to merge this code.

**Pull request merged**